File: comhash/comhash.py
import hashlib
import struct
import os
directory = os.path.dirname(os.path.realpath(__file__))

with open(os.path.join(directory, 'words.txt'), 'r') as f:
    WORDS = list(map(lambda x: x.strip().lower(), f.read().strip().split('\n')))

NUM_WORDS = 3

def comhash(secret, username):
    m = hashlib.sha256()
    m.update(username.lower().strip().encode('utf-8'))
    m.update(secret.encode('utf-8'))
    d = m.digest()
    x = struct.unpack('<%dI' % NUM_WORDS, d[0:NUM_WORDS*4])
    return ' '.join(map(lambda i: WORDS[i % len(WORDS)], x))

File: comhash/comhash.js
const crypto = require('crypto'),
      fs = require('fs');

const WORDS = fs.readFileSync(__dirname + '/words.txt', 'utf8').trim().split('\n').map((x) => x.trim().toLowerCase());

function comhash(secret, username) {
  const hash = crypto.createHash('sha256');
  hash.update(username.toLowerCase().trim() + secret, 'utf8');
  const digest = hash.digest();
  const out = [];
  for(let i = 0; i < 3; i++) {
    out.push(WORDS[digest.readUInt32LE(i*4) % WORDS.length]);
  }
  return out.join(' ');
}

module.exports = comhash;
